src/codec.rs

```diff
@@ -0,0 +1,29 @@
+//! This module contains traits to have a unified API across codecs.
+//!
+//! There are two traits defined, [`Codec`] and [`Links`]. Those are separate traits as the `Links`
+//! trait is not generic over a certain type.
+
+use cid::Cid;
+
+/// Each IPLD codec implementation should implement this Codec trait. This way codecs can be more
+/// easily exchanged or combined.
+pub trait Codec<T> {
+    /// The multicodec code of the IPLD codec.
+    const CODE: u64;
+    /// The error that is returned if encoding or decoding fails.
+    type Error;
+
+    /// Decode a slice into the desired type.
+    fn decode(bytes: &[u8]) -> Result<T, Self::Error>;
+    /// Encode a type into bytes.
+    fn encode(obj: &T) -> Result<Vec<u8>, Self::Error>;
+}
+
+/// Trait for returning the links of a serialized IPLD data.
+pub trait Links {
```

Comment:
not `Links<T>`?

Reply:
Also, I wonder if we want a relationship with `Codec<T>`? Are there any codecs where (a) we support links but (b) don't actually support enumerating them?

Reply:
It's a separate trait, so it doesn't have to be `<T>`. The idea is that you can extract links out of the encoded data. If it would be `<T>`, then it could/would be part of the `Codec` trait.

Reply:
Ah, makes sense. You don't care about T.